great_expectations/datasource/data_connector/inferred_asset_aws_glue_data_catalog_data_connector.py

```diff
@@ -0,0 +1,171 @@
+        execution_engine: Optional[ExecutionEngine] = None,
+        catalog_id: Optional[str] = None,
+        data_asset_name_prefix: str = "",
+        data_asset_name_suffix: str = "",
+        excluded_tables: Optional[list] = None,
+        included_tables: Optional[list] = None,
+        glue_introspection_directives: Optional[dict] = None,
+        boto3_options: Optional[dict] = None,
+        batch_spec_passthrough: Optional[dict] = None,
+    ):
+        """
+        DataConnector for connecting to AWS Glue Data Catalog.
+
+        Args:
+            name (str): Required name for data_connector
+            datasource_name (str): Required name for datasource
+            execution_engine (ExecutionEngine): Optional reference to ExecutionEngine
+            catalog_id (str): Optional catalog ID from which to retrieve databases. If none is provided, the AWS account ID is used by default.
+            data_asset_name_prefix (str): An optional prefix to prepend to inferred data_asset_names
+            data_asset_name_suffix (str): An optional suffix to append to inferred data_asset_names
+            excluded_tables (List): A list of tables ([database].[table]) to ignore when inferring data asset_names
+            included_tables (List): If not None, only include tables ([database].[table]) in this list when inferring data asset_names
+            glue_introspection_directives (Dict): Arguments passed to the introspection method to guide introspection
+            boto3_options (dict): optional boto3 options
+            batch_spec_passthrough (dict): dictionary with keys that will be added directly to batch_spec
+        """
+        logger.warning(
+            "Warning: great_expectations.datasource.data_connector.InferredAssetAWSGlueDataCatalogDataConnector is "
+            "experimental. Methods, APIs, and core behavior may change in the future."
+        )
+        self._data_asset_name_prefix = data_asset_name_prefix
+        self._data_asset_name_suffix = data_asset_name_suffix
+        self._excluded_tables = excluded_tables
+        self._included_tables = included_tables
+        self._glue_introspection_directives = glue_introspection_directives or {}
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            catalog_id=catalog_id,
+            assets=None,
+            batch_spec_passthrough=batch_spec_passthrough,
+            boto3_options=boto3_options,
+        )
+
+        self._introspected_assets_cache: dict = {}
+        self._refresh_introspected_assets_cache(
+            self._data_asset_name_prefix,
+            self._data_asset_name_suffix,
+            self._excluded_tables,
+            self._included_tables,
+        )
+
+    @property
+    def assets(self) -> Dict[str, dict]:
+        return self._introspected_assets_cache
+
+    def _refresh_data_references_cache(self) -> None:
```

Comment:
@lccasagrande We are very close!  Could you please mirror the style of `InferredAssetSqlDataConnector` and `ConfiguredAssetSqlDataConnector` (their updated versions have just been release!) as precisely as you can.  When you look at them, you will see significantly more code reuse (e.g., in such calls as `self._refresh_introspected_assets_cache()` and `_refresh_data_references_cache()` methods).  Thank you!

Reply:
Done, I tried to follow the Sql Data Connector style. Lemme know if something is missing or could be improved.